packages/calypso-e2e/src/lib/components/isolated-block-editor-component.ts

```diff
@@ -0,0 +1,56 @@
+import { Page, ElementHandle } from 'playwright';
+
+const selectors = {
+	blockInserterToggle: 'button[aria-label="Toggle block inserter"]',
+	blockInserterSearch: 'input[placeholder="Search"]',
+	blockInserterResultItem: '.block-editor-block-types-list__list-item',
+
+	// Publish
+	postButton: 'a:text("Post")',
+};
+/**
+ * Represents an instance of the WPCOM's Gutenberg editor page.
+ */
+export class IsolatedBlockEditorComponent {
+	private page: Page;
+
+	/**
+	 * Constructs an instance of the component.
+	 *
+	 * @param {Page} page The underlying page.
+	 */
+	constructor( page: Page ) {
+		this.page = page;
+	}
+
+	/**
+	 * Given a block name, insert a matching block to the editor.
+	 *
+	 * This method is nearly identical to the method also named `addBlock` in `GutenbergEditorPage`.
+	 * However, the major distinction is the use of `Page` vs `Frame`.
+	 *
+	 * This is because a P2 frontend (or inline) editor is not a part of an embedded iframe.
+	 *
+	 * @param {string} blockName Name of the block to be inserted.
+	 * @param {string} blockEditorSelector Unique selector to locate the inserted block in the editor.
+	 * 	Typically built into the definition file for the block being inserted.
+	 * @returns {ElementHandle} Reference to the inserted block in the editor.
+	 */
+	async addBlock( blockName: string, blockEditorSelector: string ): Promise< ElementHandle > {
+		// Click on the editor title. This has the effect of dismissing the block inserter
+		// if open, and restores focus back to the editor root container, allowing insertion
+		// of blocks.
+		await this.page.click( selectors.blockInserterToggle );
+		await this.page.fill( selectors.blockInserterSearch, blockName );
+		await this.page.click( `${ selectors.blockInserterResultItem } span:text("${ blockName }")` );
+		// Confirm the block has been added to the editor body.
+		return await this.page.waitForSelector( `${ blockEditorSelector }.is-selected` );
```

Comment:
I have identified the need to merge this with GutenbergEditorComponent, or at least create a base/superset of the two similar editor classes. 

The `addBlock` methods in both GutenbergEditorPage` and `IsolatedBlockEditorComponent` can be essentially identical, minus the use of `Page` instead of `Frame` in one case versus the other. Perhaps this is an ideal location to re-introduce (in limited capacity) the notion of inheritance?

As this is an experimental PR meant to demonstrate that we can indeed run e2e tests on P2s, I've chosen to essentially copy-paste the `addBlock` method, but once this is merged I think its' worth exploring how we can combine the two.

Reply:
`Perhaps this is an ideal location to re-introduce (in limited capacity) the notion of inheritance?`

My personal preference would be strongly to stay away from inheritance if possible! We've avoided it so far, and I just always find it makes things messier... But that's just my 0.02!

Perhaps instead we could make a `Flow` class to wrap this function, and that class takes in its constructor either a `page` or `frame`, and then the function just uses whatever was passed in in that constructor?

We could still then expose these methods on the POM classes, they would just call the shared function in the Flow class!

Reply:
> Perhaps instead we could make a Flow class to wrap this function, and that class takes in its constructor either a page or frame

That's a good idea! It would allow unifying of the API without having to alter the existing files. We'd have to update the test specs to use the new flow, but that isn't too big of a deal.

One other avenue that occurred to me is to maintain the separation. The reasoning behind this is as follows:
- IsolatedBlockEditor is maintained in a separate repository, so the POM for this should also live as a separate item.
- GutenbergEditorPage is concerned with the full-page editing experience (iframed or not), so it's a separate entity from the front-end, (slightly) simplified editing experience present on P2s.

In any case, perhaps this exploration should be done alongside adding Atomic support to GutenbergEditorPage.

Reply:
>One other avenue that occurred to me is to maintain the separation.

I think that's totally valid too!

>We'd have to update the test specs to use the new flow

Actually, you wouldn't necessarily have to! Although it's slightly inverted than usual, you could actually have the existing methods in the existing POM classes call the Flow class, and then keep the existing scripts referencing the POM classes!